### PR TITLE
Add changes from the Base API

### PR DESCRIPTION
**Pull request by notatiyyah**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-431

### *What is the problem we're trying to solve*

Since the Asset Information API was created, several updates have been made to the Base API template repository.
These changes need to be copied into the Assets Information API.

### *What changes have we introduced*

Summary of changes brought over from Base API:

#### Correlation ID support

- Added a Correlation ID Middleware
- Initialized it in startup
- Added a constants file
- Added unit tests

#### AWS XRay support

- Installed the necessary NuGet packages 
- Added requirements in Startup.cs

#### AWS DynamoDb support

- Support for setting up DynamoDb as a database, including all necessary DI registrations.
- An example Gateway implementation to show how to use the DynamoDb context interface.
- DynamoDb setup for integration tests and an example integration test file to show how it works.
- Sample Terraform files to set up a DynamoDb instance and create the necessary table(s)

#### _Checklist_

- [x] Correlation ID support
- [x] AWS XRay support
- [x] AWS DynamoDb support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,20 @@ docker run -it -v `pwd`:/app mcr.microsoft.com/powershell
 
 To serve the application, run it using your IDE of choice, we use Visual Studio CE and JetBrains Rider on Mac.
 
+**Note**
+When running locally the appropriate database conneciton details are still needed.
+##### Postgres
```
Comment (humulla) [inline README.md:58]: I would suggest to remove the postgres from the readme as we are not using it

File: database.env [hunk 2]
```diff
@@ -1,3 +1,3 @@
 POSTGRES_USER=postgres
 POSTGRES_PASSWORD=mypassword
-POSTGRES_DB=testdb
```
Comment (humulla) [inline database.env:3]: This file should be deleted

File: docker-compose.yml [hunk 3]
```diff
@@ -9,7 +9,11 @@ services:
...
     environment:
-      - CONNECTION_STRING=Host=dev-database;Port=5432;Database=testdb;Username=postgres;Password=mypassword
+     - DynamoDb_LocalMode=true
+     - DynamoDb_LocalServiceUrl=http://dynamodb-database:8000
+     - AWS_REGION=eu-west-2
+     - AWS_ACCESS_KEY_ID=local
+     - AWS_SECRET_ACCESS_KEY=local
     links:
       - dev-database
   dev-database:
```
Comment (mikemoody-amido) [inline docker-compose.yml:19]: You can remove this `dev-database` section. Then change the link above to be `dynamodb-database`.

File: docker-compose.yml [hunk 4]
```diff
@@ -9,7 +9,11 @@ services:
...
       - 3000:3000
     environment:
-      - CONNECTION_STRING=Host=dev-database;Port=5432;Database=testdb;Username=postgres;Password=mypassword
+     - DynamoDb_LocalMode=true
+     - DynamoDb_LocalServiceUrl=http://dynamodb-database:8000
+     - AWS_REGION=eu-west-2
+     - AWS_ACCESS_KEY_ID=local
+     - AWS_SECRET_ACCESS_KEY=local
     links:
       - dev-database
```
Comment (humulla) [inline docker-compose.yml:18]: This should be a link to the `dynamodb-database`

File: docker-compose.yml (hunk 3 above)
Comment (humulla) [inline docker-compose.yml:19]: This should be deleted

File: AssetsInformationApi/V1/Infrastructure/DatabaseEntity.cs [hunk 5]
```diff
@@ -19,11 +15,9 @@ namespace AssetsInformationApi.V1.Infrastructure
     [DynamoDBTable("example_table", LowerCamelCaseProperties = true)]
```
Comment (mikemoody-amido) [inline AssetsInformationApi/V1/Infrastructure/DatabaseEntity.cs:15]: You still have the `Table` class attribute there...

**Review by humulla — APPROVED** [review]

File: AssetsInformationApi/V1/Infrastructure/DatabaseEntity.cs (hunk 5 above)
Comment (notatiyyah) [reply AssetsInformationApi/V1/Infrastructure/DatabaseEntity.cs:14]: My bad, was confused because the Base API still had that line in.

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**